runtime/tools/spec-builder/assembly/index.ts

```diff
@@ -1,16 +1,48 @@
-import { UInt128 } from 'as-scale-codec';
-import { AccountData } from '@as-substrate/balances-module';
+import { UInt128, CompactInt } from 'as-scale-codec';
+import { AccountData, AccountId } from '@as-substrate/balances-module';
 import { Utils } from '@as-substrate/core-utils';
-
+import { AuraModule } from '@as-substrate/aura-module';
 
 /**
  * Gets the AccountData converted to the bytes
  * @param freeBalance free balance for the AccountData
  */
 export function getAccountDataBytes(freeBalance: Uint8Array): u8[] {
   const free = UInt128.fromU8a(Utils.toU8Array(freeBalance));
-  const accData = new AccountData(free, UInt128.Zero)
+  const accData = new AccountData(free, UInt128.Zero);
   return accData.toU8a();
 }
 
+
+/**
+ * 
+ * @param authorities list of authorities in bytes
+ */
+export function getAccountIdBytes(authorities: Uint8Array): u8[] {
+  let input = Utils.toU8Array(authorities);
+  // assumes that there are at least one AccountId in authorities
+  let counter = 1;
+  let auths: u8[] = [];
+  let accId = AccountId.fromU8Array(input);
+  auths = auths.concat(accId.result.address);
```

Comment:
I think that it will be cleaner if ] you put this into the while and start the counter from 0

Reply:
I think, suggested way would return empty `auths` array for `authorities` array with one element since condition for `while` will fail. And I can start the counter from one since I am making sure the input is not empty before providing it

Reply:
I think that if you do it this way it will work:
```JavaScript
let input = Utils.toU8Array(authorities);
let auths: u8[] = [];

while (input.length != 0) {
    accId = AccountId.fromU8Array(input);
    auths = auths.concat(accId.result.address)
    input = accId.input
    counter += 1 
}

let result: u8[] = [];
const length = new CompactInt(counter);
return result.concat(length.toU8a())
    .concat(auths);
```

but anyways